source/particle/interpolator/cell_average.cc

```diff
@@ -0,0 +1,108 @@
+                                             const std::vector<Point<dim> > &positions,
+                                             const typename parallel::distributed::Triangulation<dim>::active_cell_iterator &cell) const
+      {
+        typename parallel::distributed::Triangulation<dim>::active_cell_iterator found_cell;
+
+        if (cell == typename parallel::distributed::Triangulation<dim>::active_cell_iterator())
+          {
+            // We can not simply use one of the points as input for find_active_cell_around_point
+            // because for vertices of mesh cells we might end up getting ghost_cells as return value
+            // instead of the local active cell. So make sure we are well in the inside of a cell.
+            Point<dim> approximated_cell_midpoint = positions[0];
+            if (positions.size() > 1)
+              {
+                Tensor<1,dim> direction_to_center;
+                for (unsigned int i = 1; i<positions.size()-1; ++i)
+                  direction_to_center += positions[i] - positions[0];
+                direction_to_center /= positions.size() - 1;
+                approximated_cell_midpoint += direction_to_center;
+              }
+
+            found_cell =
+              (GridTools::find_active_cell_around_point<> (this->get_mapping(),
+                                                           this->get_triangulation(),
+                                                           approximated_cell_midpoint)).first;
+          }
+        else
+          found_cell = cell;
+
+        const types::LevelInd cell_index = std::make_pair<unsigned int, unsigned int> (found_cell->level(),found_cell->index());
+        const std::pair<typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator,
+              typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator> particle_range = particles.equal_range(cell_index);
+
+        const unsigned int n_particles = std::distance(particle_range.first,particle_range.second);
+        const unsigned int n_properties = particles.begin()->second.get_properties().size();
+        std::vector<double> cell_properties (n_properties,0.0);
+
+        AssertThrow(n_particles != 0,
+                    ExcMessage("At least one cell contained no particles. The 'constant "
+                               "average' interpolation scheme does not support this case. "));
+
+        for (typename std::multimap<types::LevelInd, Particle<dim> >::const_iterator particle = particle_range.first;
+             particle != particle_range.second; ++particle)
+          {
+            const std::vector<double> particle_properties = particle->second.get_properties();
+
+            for (unsigned int i = 0; i < n_properties; ++i)
+              cell_properties[i] += particle_properties[i] / n_particles;
+          }
+
+        const std::vector<std::vector<double> > properties(positions.size(),cell_properties);
+
+        return properties;
```

Comment:
I think it might be just as easy to not create a named variable but just return the vector right away. This won't make much of a difference in C++11 but should also be faster on older compilers.